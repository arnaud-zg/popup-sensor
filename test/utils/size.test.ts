import {
  getWindowSize,
  getCenteredBoxPosition,
  guessPopupSize,
} from '../../src/utils/size'

describe('Size', () => {
  it('should get current window size', () => {
    expect(getWindowSize()).toMatchInlineSnapshot(`
      Object {
        "height": 768,
        "width": 1024,
      }
    `)
  })

  it('should get centered box position', () => {
    expect(getCenteredBoxPosition({ height: 200, width: 200 }))
      .toMatchInlineSnapshot(`
      Object {
        "left": 412,
        "top": 284,
      }
    `)
  })

  it('should guess popup size', () => {
    expect(guessPopupSize({})).toMatchSnapshot()
    expect(guessPopupSize({ height: 200, width: 200 })).toMatchSnapshot()
    expect(guessPopupSize({ height: 1000, width: 1000 })).toMatchSnapshot()
  })
})
