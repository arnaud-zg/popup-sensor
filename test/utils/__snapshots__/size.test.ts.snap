// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Size should guess popup size 1`] = `
Object {
  "height": 384,
  "width": 512,
}
`;

exports[`Size should guess popup size 2`] = `
Object {
  "height": 200,
  "width": 200,
}
`;

exports[`Size should guess popup size 3`] = `
Object {
  "height": 384,
  "width": 512,
}
`;
